MovieGallery/Pages/Search.cshtml.cs

```diff
@@ -34,40 +36,66 @@ public void OnGet(string SearchbyYear)
             }
 
-            if (SearchbyYear == null)
-            {
-                InitialCheckList = true;
-                MoviesResult = MoviesJson;
-                ShowsResult = ShowsJson;
-            }
-            else
-            {
-                Query = SearchbyYear;
-                long inputYear = Convert.ToInt64(Query);
+            InitialCheckList = true;
+            MoviesResult = MoviesJson;
+            ShowsResult = ShowsJson;
 
-                var result = MoviesJson.Items.Where(x => x.Year == inputYear).ToList();
-                MoviesResult = new Movies()
+        }
+
+        public void OnPost(long? SearchbyYear)
+        {
+            try
+            {
+                using (StreamReader r = new StreamReader("Movies.json"))
                 {
-                    Items = result
-                };
+                    string json = r.ReadToEnd();
+                    MoviesJson = JsonConvert.DeserializeObject<Movies>(json);
+                }
 
-                var resultShow = ShowsJson.Items.Where(x => x.Year == inputYear).ToList();
-                ShowsResult = new Shows()
+                using (StreamReader r = new StreamReader("Shows.json"))
                 {
-                    Items = resultShow
-                };
+                    string json = r.ReadToEnd();
+                    ShowsJson = JsonConvert.DeserializeObject<Shows>(json);
+                }
 
-                if (MoviesResult.Items.Count > 0 || ShowsResult.Items.Count > 0)
+                if (SearchbyYear == null)
                 {
-                    SearchCompleted = true;
+                    InitialCheckList = true;
+                    MoviesResult = MoviesJson;
+                    ShowsResult = ShowsJson;
                 }
                 else
                 {
-                    SearchCompleted = false;
-                    MoviesResult = new Movies();
-                    ShowsResult = new Shows();
+
+                    var result = MoviesJson.Items.Where(x => x.Year == SearchbyYear).ToList();
+                    MoviesResult = new Movies()
+                    {
+                        Items = result
+                    };
+
+                    var resultShow = ShowsJson.Items.Where(x => x.Year == SearchbyYear).ToList();
+                    ShowsResult = new Shows()
+                    {
+                        Items = resultShow
+                    };
+
+                    if (MoviesResult.Items.Count > 0 || ShowsResult.Items.Count > 0)
+                    {
+                        SearchCompleted = true;
+                    }
+                    else
+                    {
+                        SearchCompleted = false;
+                        MoviesResult = new Movies();
+                        ShowsResult = new Shows();
+                    }
                 }
             }
+            catch
+            {
+                Console.WriteLine("An exception occured while execution");
```

Comment:
Exception blocks are good.  A few ideas to expand it:
1) Use proper logging, which has the concepts of logging levels. 
2) Give a useful indication to the user, but not too detailed.